src/dotnet/commands/dotnet-test/BaseDotnetTestRunner.cs

```diff
@@ -54,10 +53,10 @@ private int DoBuildTestProject(DotnetTestParams dotnetTestParams)
                 strings.Add(dotnetTestParams.Output);
             }
 
-            if (!string.IsNullOrEmpty(dotnetTestParams.Runtime))
+            if (!string.IsNullOrEmpty(projectContext.RuntimeIdentifier))
             {
                 strings.Add("--runtime");
-                strings.Add(dotnetTestParams.Runtime);
+                strings.Add(projectContext.RuntimeIdentifier);
```

Comment:
We create the ProjectContext based on the `runtime` and `framework` the user specified so this should reflect `--runtime`'s value.